### PR TITLE
[HDX-8403] add more logs. hail mary style

### PR DESCRIPTION
**Pull request by teodorescuserban**

### CODE REVIEW

**Comment by danmihaila** [comment]
it looks good. @teodorescuserban I am not sure about naming "KeyError" if it was on purpose or we might want to change to key_error or KEY_ERROR

**Comment by teodorescuserban** [comment]
> it looks good. @teodorescuserban I am not sure about naming "KeyError" if it was on purpose or we might want to change to key_error or KEY_ERROR

fixed.
arguably that was ok as it was representing an exception type but it's fine either way.

File: setup.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
     install_requires=[
         'requests>=2.27',
         #'libhxl==4.27',
-        'libhxl @ git+https://github.com/HXLStandard/libhxl-python.git@dev',
+        'libhxl @ git+https://github.com/HXLStandard/libhxl-python.git@hdx-8403-some-more-logs',
```
Comment (davidmegginson) [inline setup.py:31]: Need to undo this change before merging into dev.

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # libhxl==4.27
-git+https://github.com/HXLStandard/libhxl-python.git@dev#egg=libhxl
+git+https://github.com/HXLStandard/libhxl-python.git@hdx-8403-some-more-logs#egg=libhxl
```
Comment (davidmegginson) [inline requirements.txt:2]: Ditto -- need to undo this change before merging.

File: hxl_proxy/util.py [hunk 3]
```diff
@@ -217,6 +217,7 @@ def make_input_options (args):
         try:
             sheet_index = int(sheet_index)
         except:
+            logup('Invalid sheet index; defaulting to 0', {sheet_index: sheet_index}, leve='error')
```
Comment (davidmegginson) [inline hxl_proxy/util.py:220]: Typo - s/leve/level

File: hxl_proxy/controllers.py [hunk 4]
```diff
@@ -802,6 +816,7 @@ def do_data_validate():
         try:
             schema_sheet_index = int(schema_sheet_index)
         except:
+            logup('Bad schema_sheet index', {"sheet": flask.request.form.get('schema_sheet')}, 'warning')
```
Comment (davidmegginson) [inline hxl_proxy/controllers.py:819]: Didn't use level= here (probably ok).

File: hxl_proxy/controllers.py [hunk 5]
```diff
@@ -786,6 +799,7 @@ def do_data_validate():
         try:
             sheet_index = int(sheet_index)
         except:
+            logup('Bad sheet index', {"sheet": flask.request.form.get('sheet')}, 'warning')
```
Comment (davidmegginson) [inline hxl_proxy/controllers.py:802]: Didn't use level= here (probably OK).

File: hxl_proxy/controllers.py [hunk 6]
```diff
@@ -40,11 +42,15 @@ def handle_default_exception(e):
...
     """
 
+    key_error = {'error_type': type(e).__name__}
     # log a warning for the error/exception that we're handling
     if hasattr(e, 'message'):
+        key_error['message'] = e.message
         logger.warning('%s: %s', type(e).__name__, e.message)
     else:
         logger.warning(type(e).__name__)
+    logup('Error', key_error, "warning")
```
Comment (davidmegginson) [inline hxl_proxy/controllers.py:52]: Do we need to keep logger.warning/logger.info _and_ logup throughout, or would just logup be OK?

**Comment by davidmegginson** [comment]
Overall, it looks good -- see my review above. There are a few possible typos to fix, and the bigger question is whether we need to log twice (logger.info/logger.warning and logup), or just logup would be OK in those situations.

Thanks for the PR!

File: hxl_proxy/controllers.py (hunk 6 above)
Comment (teodorescuserban) [reply hxl_proxy/controllers.py:52]: I didn't want to start removing lines in the code.
I just added the json logs.
Tbh I was convinced these `logger.warning` are going to be displayed to the client in the html page at some point.

Comment (davidmegginson) [reply hxl_proxy/controllers.py:52]: OK, that's fair.
